symFileManager.py

```diff
@@ -220,3 +230,13 @@ def MaybeEvict(self, freeEntriesNeeded):
     #print "MRU after: " + str(self.sMruSymbols)
     LogDebug("Cache occupancy after MaybeEvict: " + str(self.sCacheCount) + "/" + str(maxCacheSize))
 
+  def _FetchSymbolsFromPaths(self, paths, pathSuffix, joinFunction, fetchFunction, cachePathSuffix):
```

Comment:
this function is very general in a confusing way.. 
How about this?

GetSymbols:
1. symbols = FetchSymbols()
2. Update LRU stats

FetchSymbols:
1. check in-memory cache. if found, return
2. check on-disk cache. if found, insert into memory cache and return
3. if configured, check file paths
4. if configured, check URL paths
5. if symbol found, insert into memory cache & disk cache. if symbol not found, return failure